opamp/observiq/observiq_client_test.go

```diff
@@ -173,11 +197,30 @@ func TestClientConnect(t *testing.T) {
 		{
 			desc: "TLS fails",
 			testFunc: func(*testing.T) {
+				statuses := map[string]*protobufs.PackageStatus{
+					packagestate.CollectorPackageName: {
+						Name:                 packagestate.CollectorPackageName,
+						AgentHasVersion:      version.Version(),
+						AgentHasHash:         hash,
+						ServerOfferedVersion: newVersion,
+						ServerOfferedHash:    newHash,
+						Status:               protobufs.PackageStatus_Installing,
+					},
+				}
+				packageStatuses := &protobufs.PackageStatuses{
+					ServerProvidedAllPackagesHash: allHash,
+					Packages:                      statuses,
+				}
 
 				mockOpAmpClient := new(mocks.MockOpAMPClient)
 				mockOpAmpClient.On("SetAgentDescription", mock.Anything).Return(nil)
 				mockStateProvider := new(mocks.MockPackagesStateProvider)
-				mockStateProvider.On("LastReportedStatuses").Return(nil, nil)
+				mockStateProvider.On("LastReportedStatuses").Return(packageStatuses, nil)
+				mockStateProvider.On("SetLastReportedStatuses", mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+					status := args.Get(0).(*protobufs.PackageStatuses)
+					assert.Equal(t, "Failed creating TLS config: failed to read CA file: open bad-ca.cert: no such file or directory", status.Packages[packagestate.CollectorPackageName].ErrorMessage)
```

Comment:
Also, this is failing because the specific error message for a missing file differs between Windows and linux, I think we should just assert a prefix here.